polkadot/node/core/pvf/execute-worker/src/lib.rs

```diff
@@ -16,80 +16,48 @@
 
 //! Contains the logic for executing PVFs. Used by the polkadot-execute-worker binary.
 
-pub use polkadot_node_core_pvf_common::{executor_interface::execute_artifact, worker_dir};
+pub use polkadot_node_core_pvf_common::{
+	executor_interface::execute_artifact, worker::WorkerInfo, worker_dir,
+};
 
 // NOTE: Initializing logging in e.g. tests will not have an effect in the workers, as they are
 //       separate spawned processes. Run with e.g. `RUST_LOG=parachain::pvf-execute-worker=trace`.
 const LOG_TARGET: &str = "parachain::pvf-execute-worker";
 
 use cpu_time::ProcessTime;
+#[cfg(not(target_os = "linux"))]
+use nix::unistd::ForkResult;
 use nix::{
 	errno::Errno,
 	sys::{
 		resource::{Usage, UsageWho},
 		wait::WaitStatus,
 	},
-	unistd::{ForkResult, Pid},
+	unistd::Pid,
 };
-use os_pipe::{self, PipeReader, PipeWriter};
 use parity_scale_codec::{Decode, Encode};
 use polkadot_node_core_pvf_common::{
 	error::InternalValidationError,
 	execute::{Handshake, JobError, JobResponse, JobResult, WorkerResponse},
+	executor_interface::params_to_wasmtime_semantics,
 	framed_recv_blocking, framed_send_blocking,
 	worker::{
-		cpu_time_monitor_loop, run_worker, stringify_panic_payload,
+		cpu_time_monitor_loop, pipe2_cloexec, run_worker, stringify_panic_payload,
 		thread::{self, WaitOutcome},
-		WorkerKind,
+		PipeFd, WorkerKind,
 	},
 };
 use polkadot_parachain_primitives::primitives::ValidationResult;
-use polkadot_primitives::{executor_params::DEFAULT_NATIVE_STACK_MAX, ExecutorParams};
+use polkadot_primitives::ExecutorParams;
 use std::{
 	io::{self, Read},
-	os::unix::net::UnixStream,
+	os::{fd::AsRawFd, unix::net::UnixStream},
 	path::PathBuf,
 	process,
 	sync::{mpsc::channel, Arc},
 	time::Duration,
 };
 
-// Wasmtime powers the Substrate Executor. It compiles the wasm bytecode into native code.
-// That native code does not create any stacks and just reuses the stack of the thread that
-// wasmtime was invoked from.
```

Comment:
Instead of removing this comment, can we put it above `get_max_stack_size`? It still seems relevant and useful.